src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -940,6 +941,10 @@ private void increaseLevel()
             {
                 this.levelExperienceMap.put(job.getExperienceTag(), new Tuple<>(entry.getA() + 1, entry.getB()));
             }
+            else
+            {
+                this.levelExperienceMap.put(job.getExperienceTag(),new Tuple<Integer,Double> ((int)MAX_CITIZEN_LEVEL,ExperienceUtils.getXPNeededForNextLevel(MAX_CITIZEN_LEVEL-1)));
+            }
```

Comment:
Why is this needed?

Reply:
it fixes a bug that when a guard collects too much exp it goes into infinite loop.

Reply:
Where does this infinite loop come from?

Reply:
https://github.com/ldtteam/minecolonies/blob/3f0ac6352d78fbfb93a4ec1f808c8e11152e4fea/src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenExperienceHandler.java#L131-L134

So when level up fails the loop continues, my fix is deleting the excess exp. anther idea is make all of the call tree return bool value so the loop can stop and handle the issue in the top level.

Reply:
You added * returns true if level up succeeded, but no return value btw =D

Reply:
this fix work indirectly though, the while's citizenData.getLevel queries the value and the while loop breaks

Reply:
ahhh